dmutils/request_id.py

```diff
@@ -70,19 +74,30 @@ def _get_first_header(self, header_names):
         else:
             return None
 
+    def _get_new_trace_id(self):
+        return hex(self._traceid_random.randrange(1 << 128))[2:]
+
+    def _get_new_span_id(self):
+        return hex(self._spanid_random.randrange(1 << 64))[2:]
+
     def get_onwards_request_headers(self):
         """
             Headers to add to any further (internal) http api requests we perform if we want that request to be
             considered part of this "trace id"
         """
+        new_span_id = self._get_new_span_id()
         return dict(chain(
             (
                 (header_name, self.trace_id)
                 for header_name in current_app.config['DM_TRACE_ID_HEADERS']
             ) if self.trace_id else (),
             (
-                (header_name, self.span_id)
+                (header_name, new_span_id)
                 for header_name in current_app.config['DM_SPAN_ID_HEADERS']
+            ) if self.trace_id else (),
```

Comment:
Why do we only add a span id if there's a trace id?

Reply:
Because it doesn't really make sense without a trace id - trace id's presence is kinda a signifier that zipkin is in use at all here.